src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -958,13 +958,31 @@ private static void DeleteCountOfRevisions(DocumentsOperationContext context, Sl
 
         private unsafe void MarkRevisionAsConflicted(DocumentsOperationContext context, TableValueReader tvr, Table table, Slice changeVectorSlice, Slice lowerId, Slice idSlice)
         {
+            Document revision = null;
+            if (table.IsOwned(tvr.Id) == false)
+            {
+                // We request to delete revision with the wrong collection
+                revision = TableValueToRevision(context, ref tvr);
+                var currentCollection = _documentsStorage.ExtractCollectionName(context, revision.Data);
+
+                if (_logger.IsInfoEnabled)
+                    _logger.Info($"Expected revision '{revision.Id}' with change vector '{revision.ChangeVector}' from table '{table.Name}' but revision is of collection '{currentCollection.Name}'");
+
+                table = EnsureRevisionTableCreated(context.Transaction.InnerTransaction, currentCollection);
+
+                if (table.IsOwned(tvr.Id) == false || table.ReadByKey(changeVectorSlice, out tvr) == false) // this shouldn't happened
+                    throw new VoronErrorException(
+                        $"Failed to get revision '{revision.Id}' with change vector '{revision.ChangeVector}' of collection '{currentCollection}' from table '{table.Name}'. " +
+                        "This should not happen and is likely a bug.");
+            }
```

Comment:
I saw this _almost_ exact method in various places, does it make sense to unify them?

Reply:
done